src/http-client.ts

```diff
@@ -0,0 +1,77 @@
+import crossFetch from "cross-fetch";
+
+import { HttpMethodClient, IHttpClient, IInitHttpClientOptions } from "./types";
+
+const fetch = globalThis.fetch ?? crossFetch;
```

Comment:
This [issue](https://github.com/lquixada/cross-fetch/issues/69) prevents cross-fetch to fall back to the existing global fetch implementation in Cloudflare workers, so I'm doing it manually.